.azure-pipelines.yml

```diff
@@ -0,0 +1,89 @@
+#
+# Azure Pipelines configuration for building and testing react on Linux, Windows, and macOS.
+#
+
+trigger:
+- master
+
+jobs:
+- job: LinuxBasic
+  displayName: 'Linux: Basic Tests'
+  pool:
+    vmImage: 'ubuntu-16.04'
+  steps:
+    - template: .azure-pipelines/common/setup.yml
+    - template: .azure-pipelines/linux/basic-tests.yml
+    - template: .azure-pipelines/common/publish-error-codes.yml
+      parameters:
+        artifactName: 'error-codes-linux-basic'
+
+- job: LinuxProd
+  displayName: 'Linux: Prod Tests'
+  pool:
+    vmImage: 'ubuntu-16.04'
+  steps:
+    - template: .azure-pipelines/common/setup.yml
+    - template: .azure-pipelines/linux/prod-tests.yml
+    - template: .azure-pipelines/common/publish-error-codes.yml
+      parameters:
+        artifactName: 'error-codes-linux-prod'
```

Comment:
What about just calculating the artifact name using the job name (or at least defaulting the artifact name to some version of "error-codes-{{job name}}", but still allow it to be overridden via a parameter)? This would simplify this file since you could drop all (most?) parameters/artifacts.

Reply:
I tried that...but I could seem to get it to work properly--see the artifacts for: https://dev.azure.com/kaylangan-github/react/_build/results?buildId=661.

I had defined the artifactName as:
`artifactName: join( $(Agent.JobName) , 'error-codes')`

But they all havev the JobName "Job" and it takes the `join` as part of it. Pointers?

Reply:
`$(Agent.JobIdentifier)` instead? Think this would be something like `LinuxProd`, which is probably safer. 

Note: I don't think this variable is documented ... probably should check with MattC to see if there's a reason not to use it (@apawast used it yesterday in chalk too)

Reply:
Will that resolve correctly though, or still have `join` in the name?

Reply:
The build c[hoked on that in the artifactName](https://dev.azure.com/kaylangan-github/react/_build/results?buildId=688&view=logs&jobId=e4bc3bce-8916-5372-f207-029e7f8fcf4f&taskId=da4a66fa-8fe8-5341-5202-3156cd3d59db&lineStart=9&lineEnd=10&colStart=1&colEnd=1)...but it did work for setting the `CI_PULL_REQUEST` var.

I've gone back to the parameter version for now.